### PR TITLE
remove tiflash-proxy for tiflash_compute mode

### LINKED ISSUE #6666: Remove proxy in disaggregated tiflash_compute mode
## Enhancement
tiflash_compute  node doesn't need proxy, so we can eliminate starting proxy.

### PR DESCRIPTION
**Pull request by guo-shaoge**
Signed-off-by: guo-shaoge <shaoge1994@163.com>

### What problem does this PR solve?

Issue Number: close #6666

Problem Summary:
TODO: key_manager is mock for now, we need to add new key_manager using ffi. : https://github.com/pingcap/tiflash/issues/6667

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by ywqzzy** [comment]
IMO, if we disable tiflash_proxy, we don't have to use `TiFlashProxyConfig` and `RaftStoreProxyRunner` will not be inited.

**Review by CalvinNeo — APPROVED** [review]
lgtm

File: dbms/src/Storages/Transaction/LearnerRead.cpp [hunk 1]
```diff
@@ -165,6 +165,7 @@ LearnerReadSnapshot doLearnerRead(
     const LoggerPtr & log)
 {
     assert(log != nullptr);
+    RUNTIME_CHECK(!context.isDisaggregatedComputeMode());
```
[the remark this answers is not in the document]
Comment (guo-shaoge) [reply dbms/src/Storages/Transaction/LearnerRead.cpp]: Done

File: dbms/src/Interpreters/AsynchronousMetrics.cpp [hunk 2]
```diff
@@ -121,6 +121,7 @@ static void calculateMaxAndSum(Max & max, Sum & sum, T x)
 
 FileUsageStatistics AsynchronousMetrics::getPageStorageFileUsage()
 {
+    RUNTIME_CHECK(!context.isDisaggregatedComputeMode());
```
[the remark this answers is not in the document]
Comment (guo-shaoge) [reply dbms/src/Interpreters/AsynchronousMetrics.cpp]: Done

**Review by windtalker — APPROVED** [review]
LGTM

**Review by JaySon-Huang — APPROVED** [review]
Rest LGTM

File: dbms/src/Flash/DiagnosticsService.cpp [hunk 3]
```diff
@@ -38,6 +38,12 @@ ::grpc::Status DiagnosticsService::server_info(
     ::diagnosticspb::ServerInfoResponse * response)
 try
 {
+    if (context.isDisaggregatedComputeMode())
+    {
+        String err_msg = "tiflash compute node should be managed by AutoScaler instead of PD, this grpc should not be called be AutoScaler for now";
+        LOG_ERROR(log, err_msg);
+        return ::grpc::Status(::grpc::StatusCode::INTERNAL, err_msg);
+    }
```
[the remark this answers is not in the document]
Comment (guo-shaoge) [reply dbms/src/Flash/DiagnosticsService.cpp:46]: Dashboard fetch store info from PD, after this pr, tiflash_compute will be managed by AutoScaler, so Dashboard cannot see tiflash_compute nodes.

[the remark this answers is not in the document]
Comment (JaySon-Huang) [reply dbms/src/Flash/DiagnosticsService.cpp:46]: Got it.
We'd better have a proposal on integrating tiflash_compute (and maybe auto-scaler) with dashboard in the future. Getting flame graphs from dashboard is a useful tool for development.

**Pull request merged**